### PR TITLE
feat: update email parser

### PR DESCRIPTION
**Pull request by NaMax66**
## Description
Move email parse logic to separate module

### Jira-link:
https://virtocommerce.atlassian.net/browse/ST-5389
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.44.0-pr-862-fd63-fd635bd1.zip

### CODE REVIEW

**Review by asvishnyakov — APPROVED** [review]

**Review by Andrew-Orlov — APPROVED** [review]

File: client-app/shared/company/components/invite-member-modal/invite-member-modal.vue [hunk 1]
```diff
@@ -125,20 +116,27 @@ const { value: emails } = useField<string>(
...
         },
       )
-      .matches(emailsValidationPattern, t("common.messages.invalid_value")),
+      .test(
+        "every-valid",
+        ({ value }) =>
+          t("shared.account.invite_member_dialog.invalid_emails", {
+            invalidEmails: getInvalidEmails(parseEmails(value as string)),
+          }),
+        (value) => parseEmails(value).every((el) => el.isValid),
```
Comment (vkrashenko) [inline client-app/shared/company/components/invite-member-modal/invite-member-modal.vue:138]: missing type

File: client-app/shared/company/components/invite-member-modal/invite-member-modal.vue [hunk 2]
```diff
@@ -125,20 +116,27 @@ const { value: emails } = useField<string>(
...
       )
       .test(
         "email-length",
         t("shared.account.invite_member_dialog.email_length_exceeded", { maxValue: MAX_EMAIL_LENGTH }),
-        (value: string | undefined) => {
-          const emailAddresses: string[] = getEmailAddresses(value);
-          return emailAddresses.every((emailAddress: string) => emailAddress.length <= MAX_EMAIL_LENGTH);
+        (value) => {
+          const emailAddresses = parseEmails(value);
+          return emailAddresses.every((emailAddress) => emailAddress.value.length <= MAX_EMAIL_LENGTH);
```
Comment (vkrashenko) [inline client-app/shared/company/components/invite-member-modal/invite-member-modal.vue:129]: missing type

**Review by vkrashenko — CHANGES_REQUESTED** [review]

File: client-app/shared/company/components/invite-member-modal/invite-member-modal.vue (hunk 1 above)
Comment (NaMax66) [reply client-app/shared/company/components/invite-member-modal/invite-member-modal.vue:138]: automatically evaluated by TS

File: client-app/shared/company/components/invite-member-modal/invite-member-modal.vue (hunk 2 above)
Comment (NaMax66) [reply client-app/shared/company/components/invite-member-modal/invite-member-modal.vue:129]: automatically evaluated by TS

**Review by vkrashenko — APPROVED** [review]

**Pull request merged**